ts/features/walletV3/payment/hooks/useOnTransactionActivationEffect.tsx

```diff
@@ -0,0 +1,68 @@
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import React from "react";
+import { TransactionInfo } from "../../../../../definitions/pagopa/ecommerce/TransactionInfo";
+import { TransactionStatusEnum } from "../../../../../definitions/pagopa/ecommerce/TransactionStatus";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import { walletPaymentGetTransactionInfo } from "../store/actions/networking";
+import { walletPaymentTransactionSelector } from "../store/selectors";
+import { getGenericError } from "../../../../utils/errors";
+
+const INITIAL_DELAY = 250;
+const MAX_TRIES = 3;
+
+type EffectCallback = (
+  transaction: TransactionInfo
+) => void | (() => void | undefined);
+
+/**
+ * This custom hook manages the transition of a transaction's status from ACTIVATION_REQUESTED to ACTIVATED.
+ * It employs a polling mechanism to continuously check the status, and once the status becomes ACTIVATED,
+ * the specified effect is triggered.
+ * @param effect Function to be executed upon transaction activation
+ */
+const useOnTransactionActivationEffect = (effect: EffectCallback) => {
+  const dispatch = useIODispatch();
+  const transactionPot = useIOSelector(walletPaymentTransactionSelector);
+
+  const delayRef = React.useRef(INITIAL_DELAY);
+  const countRef = React.useRef(0);
+
+  /* eslint-disable functional/immutable-data */
+  React.useEffect(() => {
+    const isMaxTriesExceeded =
+      !pot.isError(transactionPot) && countRef.current > MAX_TRIES;
+
+    if (pot.isSome(transactionPot) && !pot.isLoading(transactionPot)) {
+      const { transactionId, status } = transactionPot.value;
```

Comment:
since here the `transactionPot` could be `someError`, wouldn't it be better to use a cast to Option in order to then fold?

Reply:
Converting it to `Option` will spam the effect because `O.Some` covers the cases `Some`, `SomeLoading`, `SomeUpdating`, `SomeError`. 
So, in this case, is not safe since we need only the `Some` case.

Reply:
That said, I've refactored this part to make it clearer. Let me know! [refactor: refactored pot checking](https://github.com/pagopa/io-app/pull/5443/commits/87e420af1dfb016eee069b3ec45d8152a25347fb)